bot/exts/utilities/twemoji.py

```diff
@@ -0,0 +1,147 @@
+
+        embed = discord.Embed(
+            title=Twemoji.alias_to_name(UNICODE_EMOJI_ENGLISH[emoji]),
+            description=f"{codepoint.replace('-', ' ')}\n[Download svg]({Twemoji.get_url(codepoint, 'svg')})",
+            colour=Colours.twitter_blue,
+        )
+        embed.set_thumbnail(url=Twemoji.get_url(codepoint, "png"))
+        return embed
+
+    @staticmethod
+    def emoji(codepoint: str) -> str:
+        """
+        Returns the emoji corresponding to a given `codepoint`, or `""` if no emoji was found.
+
+        The return value is an emoji character, such as "🍂". The `codepoint`
+        argument can be of any format, since it will be trimmed automatically.
+        """
+        if code := Twemoji.trim_code(codepoint):
+            return chr(int(code, 16))
+        return ""
+
+    @staticmethod
+    def codepoint(emoji: str) -> str:
+        """
+        Returns the codepoint, in a trimmed format, of a single emoji.
+
+        `emoji` should be an emoji character, such as "🐍" and "🥰", and
+        not a codepoint like "1f1f8". When working with combined emojis,
+        such as "🇸🇪" and "👨‍👩‍👦", send the component emojis through the method
+        one at a time.
+        """
+        return hex(ord(emoji))[2:]
+
+    @staticmethod
+    def trim_code(codepoint: Optional[str]) -> Optional[str]:
+        """
+        Returns the meaningful information from the given `codepoint`.
+
+        If no codepoint is found, `None` is returned.
+
+        Example usages:
+        >>> trim_code("U+1f1f8")
+        "1f1f8"
+        >>> trim_code("\u0001f1f8")
+        "1f1f8"
+        >>> trim_code("1f466")
+        "1f466"
+        """
+        if not codepoint:
+            return None
+        if code := CODE.search(codepoint):
+            return code.group()
+
+    @staticmethod
+    def codepoint_from_input(raw_emoji: tuple[str, ...]) -> str:
+        """
+        Returns the codepoint corresponding to the passed tuple, separated by "-".
+
+        The return format matches the format used in URLs for Twemoji source files.
+
+        Example usages:
+        >>> codepoint_from_input(("🐍",))
+        "1f40d"
+        >>> codepoint_from_input(("1f1f8", "1f1ea"))
+        "1f1f8-1f1ea"
+        >>> codepoint_from_input(("👨‍👧‍👦",))
+        "1f468-200d-1f467-200d-1f466"
+        """
+        raw_emoji = [emoji.lower() for emoji in raw_emoji]
+        if is_emoji(raw_emoji[0]):
+            emojis = (Twemoji.codepoint(emoji) for emoji in raw_emoji[0])
+            return "-".join(emojis)
+
+        emoji = "".join(Twemoji.emoji(Twemoji.trim_code(code)) for code in raw_emoji)
+        if is_emoji(emoji):
+            return "-".join(Twemoji.codepoint(e) for e in emoji)
+
+        raise ValueError("No codepoint could be obtained from the given input")
+
+    @commands.command(aliases=("tw",))
+    async def twemoji(self, ctx: commands.Context, *raw_emoji: str) -> None:
```

Comment:
You need to add the correct decorators - otherwise we can't use this in dev-branding for example.

Reply:
Ah, decorators to allow it by default in all channels? Which ones do we use for that?

Reply:
It seems like [`whitelist_check`](https://github.com/python-discord/sir-lancebot/blob/main/bot/utils/decorators.py#L191) is what you're looking for.

Reply:
[`whitelist_override()`](https://github.com/python-discord/sir-lancebot/blob/main/bot/utils/decorators.py#L286-L304) is the decorator that we use for this.

Reply:
Whitelist override (the one below that) is what we actually use. On second thought, it might be best to hide the other function, but that’s a problem for later.

Because it can be confusing, what you need to do is set the role override to the everyone role. You can see an example of that here:

https://github.com/python-discord/sir-lancebot/blob/c7b93780e509e05121ce1c3deacc38c6f884a6c9/bot/exts/utilities/bookmark.py#L90

Reply:
Thanks!